x-pack/plugins/watcher/public/sections/watch_edit/components/json_watch_edit_form.tsx

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, useContext, useState } from 'react';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCodeEditor,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiLink,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { JsonWatch } from 'plugins/watcher/models/watch/json_watch';
+import { ErrableFormRow } from '../../../components/form_errors';
+import { documentationLinks } from '../../../lib/documentation_links';
+import { onWatchSave } from '../json_watch_edit_actions';
+import { WatchContext } from './watch_context';
+
+const JSON_WATCH_IDS = {
+  ID: 'watchId',
+  NAME: 'watchName',
+  JSON: 'watchJson',
+};
+
+const JSON_WATCH_ERROR_MSGS = {
```

Comment:
I think I'd rather have these defined as separate constants  with clear names (or even just inlined -- are they used in more than one spot?).  It took me a second to parse what was happening with error messages.